src/helics/application_api/Federate.cpp

```diff
@@ -834,17 +834,17 @@ void Federate::registerFilterInterfacesJson(const std::string& jsonString)
 
 void Federate::registerFilterInterfacesToml(const std::string& tomlString)
 {
-    toml::Value doc;
+    toml::value doc;
     try {
         doc = loadToml(tomlString);
     }
     catch (const std::invalid_argument& ia) {
         throw(helics::InvalidParameter(ia.what()));
     }
 
-    auto filts = doc.find("filters");
-    if (filts != nullptr) {
-        auto& filtArray = filts->as<toml::Array>();
+    if (isMember(doc, "filters")) {
+        auto filts = toml::find(doc, "filters");
+        auto& filtArray = filts.as_array();
```

Comment:
Should this have a check to make sure filts is an array?

Reply:
It didn't before, but I am not entirely clear whether everything is an array or it has to be a purposeful array in toml11.   Same comment applies to the other places

Reply:
That's an interesting question. Reading the toml spec seems to imply that arrays are values that use `[]`, but there isn't anything in the spec or toml11 that mentions implicitly conversion of a value to an array.

 Since some other places already had checks, I having it be consistent is good.

Reply:
Anyway I put in checks that throw an exception if it is not array in all the locations.  I also added a function that does the conversion to string regardless of the underlying type.  It doesn't look the toml11 does the automatic conversion.